### PR TITLE
Add openssl as optional feature

### PR DESCRIPTION
**Pull request by nacardin**

### CODE REVIEW

**Comment by sehz** [comment]
re-run CI.  Mac VM is very slow which cause issue in timing related tests

File: src/timer.rs [hunk 1]
```diff
@@ -58,35 +32,34 @@ mod test {
...
-                    // fire everytime but won't make cause more delay than initial 10 ms
-                    sleep_count += 1;
-                    debug!("timer fired");
-                }
-            }
+        while sleep_count < ITERATIONS {
+            sleep(sleep_duration).await;
+            sleep_count +=1 ;
+            debug!("timer fired, sleep_count: {}", sleep_count);
         }
```
Comment (sehz) [inline src/timer.rs:53]: this is incorrect. this will sleep for SLEEP_MS*iteration instead of just a single SLEEP_MS

**Review by sehz — CHANGES_REQUESTED** [review]
revert timer changes

File: src/openssl/async_to_sync_wrapper.rs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    pub fn unset_context(&mut self) {
+        assert_ne!(self.context, 0);
+        self.context = 0;
+    }
+
+    fn with_context<F, R>(&mut self, f: F) -> R
+    where
+        F: FnOnce(&mut Context<'_>, Pin<&mut S>) -> R,
+    {
+        unsafe {
```
Comment (sehz) [inline src/openssl/async_to_sync_wrapper.rs:41]: any work around unsafe?

File: src/timer.rs [hunk 3]
```diff
@@ -2,92 +2,11 @@ pub use inner::*;
...
-    use std::time::Duration;
-    use std::time::Instant;
-
-    use log::debug;
-    use tokio::select;
-
-    use fluvio_future::test_async;
-    use fluvio_future::timer::sleep;
-
-    /// test timer loop
```
Comment (sehz) [inline src/timer.rs:66]: why tests were removed?

**Review by sehz — CHANGES_REQUESTED** [review]

**Review by sehz — APPROVED** [review]

File: src/openssl/async_to_sync_wrapper.rs (hunk 2 above)
Comment (nacardin) [reply src/openssl/async_to_sync_wrapper.rs:41]: No alternative for this one. All we have is a pointer to work with.

**Comment by nacardin** [comment]
Let's hold off on merging until I test changes in fluvio

**Review by sehz — APPROVED** [review]
Great!

**Pull request merged**